### DISCUSSION TITLE
posts/longhorn-install/

### DISCUSSION THREAD

**Discussion by giscus** [techno-tim/techno-tim.github.io#255] [Comments]
# posts/longhorn-install/

Storage in Kubernetes is hard, complicated, and messy. Configuring volumes, mounts, and persistent volumes claims and getting it right can be a challenge. It’s also challenging to manage that storage and replicate it across all your Kubernetes clusters. It’s also been very challenging to do this on bare metal, outside of a cloud provider. That’s where Longhorn comes. Longhorn is an open source, a CNCF distributed block storage system for Kubernetes. It comes with a UI, backups, snapshots, cluster disaster recovery, and it does all this with or without Rancher. Rancher is NOT a requirement.

https://technotim.live/posts/longhorn-install/

**Comment by erktec** [comment]
I'd like to consider Longhorn for a new K3s cluster, but I can't determine in what order to install services. Would it make sense to deploy Longhorn immediately after I have my cluster configured and then deploy Traefik, Rancher, etc. after that to ensure storage for all services utilizes Longhorn for storage?
